src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/service/ApplicationService.kt

```diff
@@ -182,6 +198,16 @@ class ApplicationService(
       )
     )
 
+    managingTeamCodes.forEach {
+      createdApplication.teamCodes += applicationTeamCodeRepository.save(
+        ApplicationTeamCodeEntity(
+          id = UUID.randomUUID(),
+          application = createdApplication,
+          teamCode = it
+        )
+      )
+    }
```

Comment:
I'm curious about the state we end up in if there's a failure in storing the team codes on the previously created application. It doesn't look to me like these are wrapped up in a transaction?

Would the application be rolled back if the team codes weren't created?

If not, is it possible to send the whole package of db entities for creation to JPA together e.g. something like

```kt
val createdApplication = applicationRepository.save(
      ApprovedPremisesApplicationEntity(
        id = UUID.randomUUID(),
        crn = crn,
        createdByUser = user,
        data = null,
        document = null,
        ...
        teamCodes = teamCodes.map { buildTeamCode(it) } // something like this???
      )
    )
```

Reply:
There's a `@Transactional` annotation on `ApplicationsController::applicatonsPost` so an exception inserting the team codes will cause the Application insert to be rolled back.

Have added an integration test (`Create new application returns 500 and does not create Application without team codes when write to team code table fails`) to prove this.

Reply:
ah, interesting, thank you!